src/screenshot_gui.cpp

```diff
@@ -50,31 +45,11 @@ struct ScreenshotWindow : Window {
 	/**
 	 * Make a screenshot.
 	 * Ask for confirmation if the screenshot will be huge.
-	 * @param t Screenshot type: World, defaultzoom, heightmap or viewport screenshot
-	 */
-	static void TakeScreenshot(ScreenshotType st) {
-		ViewPort vp;
-		SetupScreenshotViewport(st, &vp);
-		if ((uint64)vp.width * (uint64)vp.height > 8192 * 8192) {
-			/* Ask for confirmation */
-			_screenshot_type = st;
-			SetDParam(0, vp.width);
-			SetDParam(1, vp.height);
-			ShowQuery(STR_WARNING_SCREENSHOT_SIZE_CAPTION, STR_WARNING_SCREENSHOT_SIZE_MESSAGE, nullptr, ScreenshotConfirmationCallback);
-		}
-		else {
-			/* Less than 64M pixels, just do it */
-			MakeScreenshot(st, nullptr);
-		}
-	}
-
-	/**
-	 * Callback on the confirmation window for huge screenshots.
-	 * @param w Window with viewport
-	 * @param confirmed true on confirmation
+	 * @param st Screenshot type: World, defaultzoom, heightmap or viewport screenshot
 	 */
-	static void ScreenshotConfirmationCallback(Window *w, bool confirmed) {
-		if (confirmed) MakeScreenshot(_screenshot_type, nullptr);
+	static void TakeScreenshot(ScreenshotType st)
```

Comment:
Same as previous, no need for this function to exist anymore - just call the other one

Reply:
Is it even called anywhere anyway?

Reply:
> Same as previous, no need for this function to exist anymore - just call the other one

As you wish. Should `MenuClickScreenshot` in `toolbar_gui.cpp` get the same treatment? If not, then why not?

> Is it even called anywhere anyway?

It's called in the line immediately above, when the user clicks the button.

Reply:
Yeah, I guess it should

Reply:
I am still utterly befuddled as to how and why we ended up with two identical implementations of `TakeScreenshot`; especially given that one has clearly had identifiers renamed so as to avoid conflict with the other.